onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -865,6 +866,36 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
   auto& ep_graph = ep_context_model.MainGraph();
   ep_graph.SetDescription(graph.Description());
 
+  // Generate EP compatibility strings for OrtEp types and add to model metadata
+  {
+    const GraphViewer graph_viewer(graph);
+    for (const auto& ep : execution_providers) {
+      try {
+        // Generate the compatibility string for this EP
+        std::string compatibility_string = ep->GetCompiledModelCompatibilityInfo(graph_viewer);
+        if (!compatibility_string.empty()) {
+          // Create a unique key for this EP's compatibility info
+          // Use format: "ep_compatibility_info.<EP_TYPE>"
+          // TODO: Will this guarantee uniqueness across all EPs?
+          std::string metadata_key = std::string(kOrtModelMetadata_EpCompatibilityInfoPrefix) + ep->Type();
+          auto& model_metadata = ep_context_model.MetaData();
+          auto [it, was_inserted] =
+              model_metadata.insert_or_assign(metadata_key, compatibility_string);
+          if (!was_inserted) {
+            LOGS(logger, WARNING) << "Overwriting existing EP compatibility info for key: " << metadata_key << " (EP: " << ep->Type() << ")";
+          }
+          LOGS(logger, INFO) << "Added EP compatibility info for " << ep->Type() << " with key: " << metadata_key;
+
+        } else {
+          LOGS(logger, WARNING) << "EP " << ep->Type() << " returned empty compatibility string";
```

Comment:
nit: because an empty compatibility string is the default behavior, perhaps we don't need to log this?